experiments/add_sf_to_df_add_row.R

```diff
@@ -350,6 +390,34 @@ geo_attributes <- function(dat, zoomto = NULL){
     # provide mechanism to return after all done
     observeEvent(input$donebtn, {
 
+      if(grepl(class(df$data$geometry)[[1]], "sfc_GEOMETRY")){
+
+        if (any(st_is_empty(df$data$geometry))) {
+          shinyWidgets::show_alert('Missing Geometry',
+                                   'some features are missing geometry, these must be entered before saving',
+                                   type = 'warning')
+        } else {
+          stopApp({
+
+        geo_type <- lapply(df$data$geometry, FUN = function(x){class(x)[[2]]})
+
+        df_geo_type <- rbind.data.frame(matrix(unlist(geo_type), nrow=length(geo_type), byrow=TRUE,dimnames = NULL))
+
+        colnames(df_geo_type)[[1]] <- 'geo_type'
+
+        df_out_list <- cbind.data.frame(df_geo_type, df$data)
+        df_out_list <- st_sf(df_out_list, crs = APP_CRS)
+        out <- split(df_out_list , f = df_out_list$geo_type)
+
+        # clean bounding box just in case
+        #attr(st_geometry(df_out_list), "bbox") <- st_bbox(st_union(df_out_list$geometry))
```

Comment:
this line needs to remain. when adding row by row the bbox gets messed up. perhaps it can go to the start of the `stopApp({})` function